node/resolvers/Routes/queries.ts

```diff
@@ -1,6 +1,6 @@
 export const QUERIES = {
   getPermission: `query permissions {
-    checkUserPermission @context(provider: "vtex.storefront-permissions"){
```

Comment:
It's generally better to keep this, as it avoids conflicts with other installed apps that may define a query with the same name. Is there a reason for removing it?

Reply:
Yes I got the conflicts error before in another test too. But in my tests now, I have some errors when I let this context. [log](https://developer.logs.vtex.com/_dashboards/goto/780bb96f2bbccda1a3f002c3481380d7?security_tenant=global)

![image](https://github.com/vtex-apps/b2b-checkout-settings/assets/5332361/a0efb812-796d-4edf-9820-9359fb4119b9)

Reply:
Oh got it, this new type of client is calling the apps directly, without graphql-server in the middle. It should be fine, just not the usual flow.

Reply:
This is my doubt, what is the normal flow? Call `graphql-server` >> `app-graphql` or call directly. I saw the many ways in other apps, but for me, it makes more sense to call directly or we can have SPOFs(Single Point of Failure) and I imagined it was necessary to scale `graphql-server` more than necessary. Do you know what the objective of `graphql-server` is?

Reply:
The reason it exists is to provide a full GraphQL API for apps, allowing callers to request multiple queries from different apps in the same request, for example. Graphql-server is basically a gateway, redirecting each part of the original request to the right graphql app. It also allows graphql apps to extend types from other apps, adding fields. Handling those extended types can only be done in such a gateway, so if B2B Suite used this we wouldn't be able to call an app directly.

But B2B Suite doesn't use this anywhere as far as I've seen. So here `graphql-server` only adds to the latency, like you said, and it's ok to call the apps directly. But in general the recommended pattern is to always go through graphql-server so that the API can be used with all its functionalities.